bundles/org.openhab.binding.phc/src/main/java/org/openhab/binding/phc/internal/handler/PHCBridgeHandler.java

```diff
@@ -104,215 +110,405 @@ public void initialize() {
+            do {
+                switch (qo.getModuleType()) {
+                    case PHCBindingConstants.CHANNELS_AM:
+                        sendAm(qo.getModuleAddress(), qo.getChannel(), qo.getCommand());
+                        break;
+                    case PHCBindingConstants.CHANNELS_EM_LED:
+                        sendEm(qo.getModuleAddress(), qo.getChannel(), qo.getCommand());
+                        break;
+                    case PHCBindingConstants.CHANNELS_JRM:
+                        sendJrm(qo.getModuleAddress(), qo.getChannel(), qo.getCommand(), qo.getTime());
+                        break;
+                    case PHCBindingConstants.CHANNELS_DIM:
+                        sendDim(qo.getModuleAddress(), qo.getChannel(), qo.getCommand(), qo.getTime());
+                        break;
+                }
 
-                    queue.put(System.currentTimeMillis() + SEND_RETRY_TIME_MILLIS, qo);
+                sendCount++;
 
-                } else if (qo.getCounter() >= SEND_RETRY_COUNT
-                        && !qo.getModuleType().equals(PHCBindingConstants.CHANNELS_JRM)) {
-                    // Can´t process the acknowledgement of JRM yet.
-                    logger.info("No acknowlgdge from the module {} received.", qo.getModuleAddress());
+                try {
+                    Thread.sleep(SEND_RETRY_TIME_MILLIS);
+                } catch (InterruptedException e) {
+                    Thread.currentThread().interrupt();
                 }
+            } while (!isChannelOutputState(qo.getModuleType(), qo.getModuleAddress(), qo.getChannel(), qo.getCommand())
+                    && sendCount < SEND_RETRY_COUNT);
+
+            if (qo.getModuleType().equals(PHCBindingConstants.CHANNELS_JRM)) {
+                // there aren't state per channel for JRM modules
+                amOutputState[qo.getModuleAddress() & 0x1F] = -1;
+            } else if (qo.getModuleType().equals(PHCBindingConstants.CHANNELS_DIM)) {
+                // state ist the same for every dim level except zero/off -> inizialize state
+                // with 0x0F after sending an command.
+                dmOutputState[qo.getModuleAddress() & 0x1F] |= (0x0F << (qo.getChannel() * 4));
             }
 
+            if (sendCount >= SEND_RETRY_COUNT) {
+                // change the toggle: if no acknowledge received it may be wrong.
+                byte module = qo.getModuleAddress();
+                if (PHCBindingConstants.CHANNELS_AM.equals(qo.getModuleType())
+                        || PHCBindingConstants.CHANNELS_JRM.equals(qo.getModuleType())) {
+                    module |= 0x40;
+                } else if (PHCBindingConstants.CHANNELS_DIM.equals(qo.getModuleType())) {
+                    module |= 0xA0;
+                }
+                toggleMap.put(module, !getToggle(module));
+
+                if (logger.isDebugEnabled()) {
```

Comment:
Should we update the Thing status?

Reply:
We could set the status to UNKNOWN, but then we would have to check the status after each successful send and set it back to ONLINE.
What would be the right/desired way, since updateStatus (protected) is not directly accessible.

Reply:
You don't have to check the status, you can just update it to ONLINE after every successful send that is a common and best-practice pattern.